src/mscorlib/src/System/String.cs

```diff
@@ -159,27 +159,26 @@ public sealed class String : IComparable, ICloneable, IConvertible, IEnumerable
             Contract.Ensures(Contract.Result<String>() != null);
             Contract.EndContractBlock();
 
-            if (separator == null)
```

Comment:
Makes sense. As long as we document somewhere that this was an intentional change to remove the null check :smile: